Runtime/UI/SafeArea.cs

```diff
@@ -0,0 +1,95 @@
+using UnityEngine;
+
+namespace Graphy.Runtime.UI
+{
+    /// <summary>
+    ///     Component that matches the size of the RectTransform to the safe area.
+    /// </summary>
+    [ExecuteAlways]
+    public sealed class SafeArea : MonoBehaviour
+    {
+        private Canvas m_canvas;
+        private bool m_isInitialized;
+        private Rect m_latestSafeArea;
+        private RectTransform m_rectTransform;
+        private DrivenRectTransformTracker m_rectTransformTracker;
+
+        private RectTransform RectTransform
+        {
+            get
+            {
+                if (m_rectTransform == null)
+                    m_rectTransform = (RectTransform)transform;
+
+                return m_rectTransform;
+            }
+        }
+
+        private Canvas Canvas
+        {
+            get
+            {
+                if (m_canvas == null)
+                    m_canvas = GetComponentInParent<Canvas>();
+
+                return m_canvas;
+            }
+        }
+
+        private void Update()
+        {
+            var changed = Apply();
+            if (changed)
```

Comment:
Not a huge fan of checking this every frame. I don't mind it being an option, but in reality most devices safe areas don't change, we basically have to do this once on boot and that's it. We could get rid of Update entirely and just do this once in Start for example. What do you think?

Reply:
can do this when [OnRectTransformDimensionChange](https://docs.unity3d.com/2018.4/Documentation/ScriptReference/EventSystems.UIBehaviour.OnRectTransformDimensionsChange.html) is called

Reply:
Thanks for the good point. In mobile apps, the safe area changes when the screen orientation is switched, so it needs to be checked for apps that don't have a fixed orientation. However, there are many apps that fix it, so I made it an option and set the default false.

Note that it is not applied in Edit Mode because it must always be checked.

Reply:
`OnRectTransformDimensionsChange` is the method of the UIBehaviour, so I don't use it in this case.